### PR TITLE
extract accepted cipher list and expose it

### PR DESCRIPTION
**Pull request by buckley-w-david**
This PR extracts the cipher list from the sslyze output and verifies that all accepted ciphers satisfy the minimum security requirements.

### CODE REVIEW

File: tracker/data/processing.py [hunk 1]
```diff
@@ -36,6 +36,51 @@
 SCAN_CACHE = os.path.join(env.SCAN_DATA, "cache")
 SCAN_DOMAINS_CSV = os.path.join(SCAN_CACHE, "domains.csv")
 
+ACCEPTED_CIPHERS = {
```
Comment (obrien-j) [inline tracker/data/processing.py]: Is there a way to pull these out into a config.py or equivalent?

File: tracker/data/processing.py [hunk 2]
```diff
@@ -36,6 +36,51 @@
...
+    "TLS_RSA_WITH_AES_256_CCM",
+    "TLS_RSA_WITH_AES_128_CBC_SHA256",
+    "TLS_RSA_WITH_AES_256_CBC_SHA256",
+    "TLS_RSA_WITH_AES_128_CBC_SHA",
+    "TLS_RSA_WITH_AES_256_CBC_SHA",
+    "TLS_RSA_WITH_3DES_EDE_CBC_SHA",
+    "TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA",
+    "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA",
+    "TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA",
+    "TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA.",
```
Comment (obrien-j) [inline tracker/data/processing.py]: Have we gotten confirmation that these last few are legit? They seem to be ones I would expect to be deprecated.

File: tracker/data/processing.py [hunk 3]
```diff
@@ -573,11 +619,15 @@ def https_behavior_for(name, pshtt, sslyze, parent_preloaded=None):
         else:
             bod_crypto = 1
 
+
+        uses_accepted_ciphers = all(cipher in ACCEPTED_CIPHERS for cipher in sslyze.get("Accepted Ciphers").split(', '))
```
Comment (obrien-j) [inline tracker/data/processing.py]: This is fine, just curious if there's a way to also populate the list of 'unapproved ciphers', even if it's just something we dump out via the CSV export or something. I'd rather have more data in the DB then less.

File: tracker/data/processing.py (hunk 2 above)
Comment (buckley-w-david) [reply tracker/data/processing.py]: You can go look at Tim's comment on the Trello card, but it seemed to me to say that all the ones in the list were accepted. It is ordered by their preference for which is used, so it makes sense that the last few are the most iffy.

What do you think? Keep them or boot them from the list?

File: tracker/data/processing.py (hunk 3 above)
Comment (buckley-w-david) [reply tracker/data/processing.py]: Yeah, that shouldn't be difficult. I can make a change to that effect

File: tracker/data/processing.py (hunk 1 above)
Comment (buckley-w-david) [reply tracker/data/processing.py]: Still relevant if we go dynamic from DB?

File: tracker/data/processing.py (hunk 2 above)
Comment (ptd-tbs) [reply tracker/data/processing.py]: Those need to be booted from the list please.

**Pull request closed without merging**